src/crush/CrushWrapper.h

```diff
@@ -986,10 +986,10 @@ class CrushWrapper {
     if (IS_ERR(r)) return PTR_ERR(r);
     return r->len;
   }
-  int get_rule_mask_ruleset(unsigned ruleno) const {
+  int get_rule_mask_rule_id(unsigned ruleno) const {
     crush_rule *r = get_rule(ruleno);
     if (IS_ERR(r)) return -1;
-    return r->mask.ruleset;
+    return r->mask.rule_id;
```

Comment:
This whole method is unnecessary... since rule_id always == the position.  Better to remove it, I think, and shake out any cruft callers who are trying to cope with them being different!

Reply:
Actually, I think a better approach would be to keep the 'ruleset' naming and member as a signal that this is a legacy, unused field.  If we backport this much to luminous, then the next path (not backported) can completely remove the field and all remaining traces of "ruleset".